impl/src/main/java/eu/darkbot/impl/managers/AbstractAttackImpl.java

```diff
@@ -34,6 +34,11 @@ protected AbstractAttackImpl(HeroItemsAPI heroItems, HeroAPI hero) {
 
     @Override
     public void setTarget(@Nullable Lockable target) {
+        // target changes so disarm timers
+        if (this.target != target) {
+            lockTry.disarm();
+            attackTry.disarm();
+        }
         this.target = target;
```

Comment:
```suggestion
        if (this.target == target) return;
        this.target = target;
        lockTry.disarm();
        attackTry.disarm();
```
I think it may be simpler like this